includes/utilities.inc

```diff
@@ -304,6 +304,7 @@ function islandora_solr_prepare_solr_results($solr_results) {
 
     foreach ($solr_doc as $field => $original_value) {
       $value = (array) $original_value;
+      $separator = array_key_exists($field, $truncate_length) ? variable_get('islandora_solr_search_truncated_field_value_separator', "\n") : variable_get('islandora_solr_search_field_value_separator', ', ');
```

Comment:
Think it makes more sense to use the `variable_get`s in the spots where they are used as opposed to some dynamic checking to set the seperator as it's a little harder to read / follow without intricate knowledge.